subsys/testsuite/ztest/include/ztest_test_new.h

```diff
@@ -114,6 +114,13 @@ extern struct ztest_suite_node _ztest_suite_node_list_end[];
 		.predicate = PREDICATE,							\
 	}
 
+/**
+ * Default entry point for running or listing registered unit tests.
+ *
+ * @param state The current state of the machine as it relates to the test executable.
+ */
+void ztest_run_all(const void *state);
```

Comment:
I think there's a small mixed with the `ztest_arch_api` here. I would assume that this is intended to be the `run_all` from the API? Probably want to remove this from the header and make the implementations of this static so they're only accessible from the API (unless I didn't understand something)

Reply:
Updated per offline discussion